File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.go
func hasSameDigits(s string) bool {
	last := make([]int, 0, 100)
	for i := 0; i < len(s)-1; i++ {
		last = append(last, (int(s[i]-'0')+int(s[i+1]-'0'))%10)
	}

	for len(last) > 2 {
		curr := make([]int, 0, 100)
		for i := 0; i < len(last)-1; i++ {
			curr = append(curr, (last[i]+last[i+1])%10)
		}
		last = curr
	}

	return last[0] == last[1]
}

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.cpp
#pragma GCC optimize("O3", "unroll-loops");
static const int __ = []() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  return 0;
}();
class Solution {
 public:
  bool hasSameDigits(string s) {
    int last[100];
    for (int i = 0; i < s.size() - 1; i++) {
      last[i] = (((s[i] - '0') + (s[i + 1] - '0')) % 10);
    }

    int cnt = s.size() - 1;
    while (cnt > 2) {
      int curr[100];
      for (int i = 0; i < cnt - 1; i++) {
        curr[i] = ((last[i] + last[i + 1]) % 10);
      }
      memcpy(last, curr, sizeof(curr));
      cnt--;
    }

    return last[0] == last[1];
  }
};